File: modules/dxp/apps/osb/osb-asah/osb-asah-stream-curator/src/main/java/com/liferay/osb/asah/stream/curator/model/journal/Journal.java
/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * The contents of this file are subject to the terms of the Liferay Enterprise
 * Subscription License ("License"). You may not use this file except in
 * compliance with the License. You can obtain a copy of the License by
 * contacting Liferay, Inc. See the License for the specific language governing
 * permissions and limitations under the License, including but not limited to
 * distribution rights of the Software.
 *
 *
 *
 */

package com.liferay.osb.asah.stream.curator.model.journal;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonTypeInfo;

import com.liferay.osb.asah.stream.curator.model.BaseAssetModel;

/**
 * @author Inácio Nery
 */
@JsonIgnoreProperties(ignoreUnknown = true)
@JsonTypeInfo(
	defaultImpl = Journal.class, include = JsonTypeInfo.As.EXISTING_PROPERTY,
	property = "messageFormat", use = JsonTypeInfo.Id.NAME, visible = true
)
public class Journal extends BaseAssetModel {

	public void addViews(long views) {
		_views += views;
	}

	public long getRatings() {
		return _ratings;
	}

	public float getRatingsScore() {
		return _ratingsScore;
	}

	public long getViews() {
		return _views;
	}

	public void setRatings(long ratings) {
		_ratings = ratings;
	}

	public void setRatingsScore(float ratingsScore) {
		_ratingsScore = ratingsScore;
	}

	public void setViews(long views) {
		_views = views;
	}

	private long _ratings;
	private float _ratingsScore;
	private long _views;

}

File: modules/dxp/apps/osb/osb-asah/osb-asah-spark-python/src/main/python/liferay/stream/processor/journal.py
#
# Copyright (c) 2000-present Liferay, Inc. All rights reserved.
#
# The contents of this file are subject to the terms of the Liferay Enterprise
# Subscription License ("License"). You may not use this file except in
# compliance with the License. You can obtain a copy of the License by
# contacting Liferay, Inc. See the License for the specific language governing
# permissions and limitations under the License, including but not limited to
# distribution rights of the Software.
#

from liferay.stream.processor.common import AnalyticsEventsDataFrameProcessor

from pyspark.sql import functions as F

class JournalDataFrameProcessor(AnalyticsEventsDataFrameProcessor):

	def _filter(self, analytics_events_data_frame):
		return analytics_events_data_frame.filter(
			"""
				(applicationId = 'WebContent') AND
				(eventId = 'webContentViewed') AND
				(eventProperties.articleId != '')
			"""
		)

	def _get_asset_id_column(self):
		return F.col('eventProperties.articleId')

	def _process(self, data_frame):
		return data_frame.withColumn(
			'views', F.lit(1)
		).groupBy(
			'projectId', 'channelId', 'userId', 'assetId', 'variantId',
			'normalized_event_date'
		).agg(
			F.sum('views').alias('views')
		)